### PR TITLE
Mdns: move Hostname into MdnsService struct

### PR DESCRIPTION
**Pull request by cecille**
#### Problem
The commissionable node discovery needs a place to return the host name, but hostname is currently separate from the MdnsService structure.

#### Change overview
Moves the host name into the MdnsServices structure. Also removes the separate calls to set host name.

#### Testing
covered in part by the TestMdns test. Tested on M5 and linux using chip-device-ctrl compiled with chip_mdns=platform.

### CODE REVIEW

**Review by msandstedt — APPROVED** [review]

**Review by bzbarsky-apple — APPROVED** [review]

File: src/platform/Linux/MdnsImpl.cpp [hunk 1]
```diff
@@ -761,13 +761,12 @@ CHIP_ERROR ChipMdnsInit(MdnsAsyncReturnCallback initCallback, MdnsAsyncReturnCal
...
-CHIP_ERROR ChipMdnsSetHostname(const char * hostname)
-{
-    return MdnsAvahi::GetInstance().SetHostname(hostname);
-}
-
 CHIP_ERROR ChipMdnsPublishService(const MdnsService * service)
 {
+    if (strcmp(service->mHostName, "") != 0)
+    {
+        MdnsAvahi::GetInstance().SetHostname(service->mHostName);
```
Comment (Damian-Nordic) [inline src/platform/Linux/MdnsImpl.cpp:768]: check for error?

File: src/platform/OpenThread/MdnsImpl.cpp [hunk 2]
```diff
@@ -47,6 +42,10 @@ CHIP_ERROR ChipMdnsPublishService(const MdnsService * service)
     CHIP_ERROR result = CHIP_NO_ERROR;
 
     VerifyOrExit(service, result = CHIP_ERROR_INVALID_ARGUMENT);
+    if (strcmp(service->mHostName, "") != 0)
+    {
+        ThreadStackMgr().SetupSrpHost(service->mHostName);
```
Comment (Damian-Nordic) [inline src/platform/OpenThread/MdnsImpl.cpp:47]: check for error?

File: src/platform/Linux/MdnsImpl.cpp (hunk 1 above)
Comment (cecille) [reply src/platform/Linux/MdnsImpl.cpp]: added in next commit

File: src/platform/OpenThread/MdnsImpl.cpp (hunk 2 above)
Comment (cecille) [reply src/platform/OpenThread/MdnsImpl.cpp]: Added in next commit

**Review by mspang — APPROVED** [review]

**Pull request merged**